### PR TITLE
Update ehighways resolution shape generation

### LINKED ISSUE #318: Build ehighways resolution units
### What can be improved?

Take regional level NUTS3 data using the correct NUTS year and aggregate up to eHighways resolution. See "data/statistical_units_to_ehighways_regions.csv" in sector-coupled repo for aggregation mapping

### Version

1.2.0.dev

### PR DESCRIPTION
**Pull request by brynpickering**
Supercedes #336
Fixes #318 

This is in response to a question you had @timtroendle around defining the shape source in `shapes` in the config. I went to clean up the unresolved comments from #336 and realised that it is a lot cleaner to take that approach. This PR leads to a less flexible solution - E-Highways is baked in more than in #336 - but it is cleaner.

As with #336, it only works with NUTS2013, but I will open an issue about that and have a fix ready to go that I'll add via another PR.
 
## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [x] CHANGELOG updated
- [x] Minimal workflow tests pass
- [x] Tests added to cover contribution
- [x] Documentation updated
- [x] Configuration schema updated

### CODE REVIEW

File: scripts/shapes/units.py [hunk 1]
```diff
@@ -3,95 +3,145 @@
...
-    ), f"Countries are missing in layer {layer_name}."
+def _iso3(country_name: str) -> str:
+    return utils.convert_country_code(country_name, output="alpha3")
 
 
-def _iso3(country_name):
-    return pycountry.countries.lookup(country_name).alpha_3
+def _continental_layer(units: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    "For the continental resolution, dissolve all"
+    #
```
Comment (timtroendle) [inline scripts/shapes/units.py:107]: Remove

File: scripts/shapes/units.py [hunk 2]
```diff
@@ -3,95 +3,145 @@
...
-    assert all(
-        _iso3(country) in layer.country_code.unique() for country in countries
-    ), f"Countries are missing in layer {layer_name}."
+def _iso3(country_name: str) -> str:
+    return utils.convert_country_code(country_name, output="alpha3")
 
 
-def _iso3(country_name):
-    return pycountry.countries.lookup(country_name).alpha_3
+def _continental_layer(units: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
```
Comment (timtroendle) [inline scripts/shapes/units.py:105]: Consider renaming to `merge_to_continental_layer` or similar.

**Review by timtroendle — APPROVED** [review]
I am liking this a lot. Just a few minor comments.

**Review by timtroendle — CHANGES_REQUESTED** [review]
CHANGELOG and documentation updates are missing.

File: config/schema.yaml [hunk 3]
```diff
@@ -517,6 +521,7 @@ properties:
                                 - "Norway"
                                 - "Serbia"
                                 - "Switzerland"
+                                - "Iceland" # added for ehighways
```
[the remark this answers is not in the document]
Comment (brynpickering) [reply config/schema.yaml:524]: sorry, slipped through the net

File: envs/default.yaml [hunk 4]
```diff
@@ -16,7 +16,7 @@ dependencies:
     - netCDF4=1.5.6
     - hdf5=1.10
     - h5py=3.1.0
-    - libnetcdf=4.7
+    - libnetcdf=4.8.0
```
[the remark this answers is not in the document]
Comment (brynpickering) [reply envs/default.yaml:19]: I don't have a linux device to try it on, so we'll have to wait for one of the others

**Comment by brynpickering** [comment]
@timtroendle I've addressed your issues but...

On running the minimal workflow with updated zenodo pointer (to new potentials data that includes `ehighways`), a previous issue rears its head (#99). This would be solved by merging in #239.

**Comment by timtroendle** [comment]
Will have a look at #239 soon.

**Comment by brynpickering** [comment]
Minimal workflow now runs successfully, including `all_tests` (which covers the `ehighways` resolution as well as the `regional` one).

File: config/schema.yaml [hunk 5]
```diff
@@ -396,6 +396,10 @@ properties:
                 type: string
                 description: Nuclear technology capacity scenario. "current" will use today's capacities based on the JRC power plant database (PPDB). All other scenario names should point to a configuration CSV file specifying minimum and maximum capacities per country in MW with non-zero capacities (columns=[country, min, max]).
                 pattern: ^(current)|(\w+.csv)$
+            nuts-year:
+                description: Version of the Nomenclature of Territorial Units for Statistics to use in creating spatial units (see https://ec.europa.eu/eurostat/web/nuts/history for more info).
+                type: number
+                enum: [2013]  # FIXME: enable other years
```
Comment (timtroendle) [inline config/schema.yaml:402]: Doesn't sound like a fixme to me.

**Review by timtroendle — APPROVED** [review]

**Pull request merged**